### ISSUE TITLE
filesrc not working with interpipe

### ISSUE THREAD

**Issue by anilgond** [RidgeRun/gst-interpipe#55]
gstc_pipeline_create (client, "pipe1", "filesrc location=/home/1.mpg ! decodebin ! videoconvert ! capsfilter caps=video/x-raw,format=(string)I420,width=640,height=480,framerate=(fraction)30/1 ! interpipesink name=src_1 sync=false async=false");

gstc_pipeline_create (client, "pipe_sink", "interpipesrc name=interpipesrc1 listen-to=src_1 ! queue ! capsfilter caps=video/x-raw,format=(string)I420,width=[300,640],height=[200,480],framerate=(fraction)30/1 !  autovideosink async=false sync=false");

I have tried multiple pipelines for filesrc with interpipe but it didn't worked, what I am doing incorrectly.

No error in gstd but no output.

While building interpipe, make check failed 2 test (already reported on github) is it related to filesrc?
Any help will be appreciated.
Thanks in advance.

**Comment by jcaballeros** [comment]
Hi @anilgond,

We are looking into those failed tests and will be resolved soon, however, I don't think that those are related to your problem.

Can you confirm that the pipeline without the interpipes works with gst-launch? Please run:

`gst-launch-1.0 filesrc location=/home/1.mpg ! decodebin ! videoconvert ! capsfilter caps=video/x-raw,format=(string)I420,width=640,height=480,framerate=(fraction)30/1 ! queue ! capsfilter caps=video/x-raw,format=(string)I420,width=[300,640],height=[200,480],framerate=(fraction)30/1 ! autovideosink async=false sync=false`

Also, which platform, gstd version and gst-interpipe version are you using?

**Comment by anilgond** [comment]
Hi @jcaballeros thanks for your reply,

Above pipeline didn't worked with caps, without caps its working fine. If I don't use caps with interpipe, then gstd will give error "caps are not defined".
After adding caps gstd will not throw any error but also there is no output.

I am working on Centos 7, gstd-1.0, interpipe 1.1.0.

**Comment by anilgond** [comment]
Hi @jcaballeros thanks for your reply,
I am working on Centos 7, gstd-1.0, interpipe 1.1.0.

I have corrected the caps and above pipeline is working.

But still interpipes are not working,  now I can see the output but it didn't changed using listen-to property, only first video play. Following are my pipelines.
 
Pipe 1:
filesrc location=/home/1.mpg ! decodebin ! videoconvert ! video/x-raw,width=720,height=576,format=I420,framerate=25/1 ! queue ! interpipesink name=src_3 sync=false async=false

Pipe 2:
filesrc location=/home/2.mpg ! decodebin ! videoconvert ! video/x-raw,width=720,height=576,format=I420,framerate=25/1 ! queue ! interpipesink name=src_3 sync=false async=false

Sink pipe:
interpipesrc name=interpipesrc1 listen-to=src_1 allow-renegotiation=true ! queue ! autovideosink async=false sync=false

**Comment by jcaballeros** [comment]
Hi @anilgond,

I'd suggest sending an eos event to the pipe1, and stopping pipe3 before switching to listen-to pipe2, for instance:

```
event_eos pipe1
pipeline_stop pipe3
pipeline_stop pipe1
element_set pipe3 interpipesrc1 listen-to src_2
pipeline_play pipe3
pipeline_play pipe2
```

Also, please enable GStreamer debug level warning to get some information of what's failing in the switch, check: [GStreamer_Daemon_-_Enabling_the_Debug_Subsystem](https://developer.ridgerun.com/wiki/index.php?title=GStreamer_Daemon_-_Enabling_the_Debug_Subsystem)

**Comment by jcaballeros** [comment]
Closing issue. Feel free to reopen if necessary.

**Comment by anilgond** [comment]
Its not working yet, I have tried above steps but I am getting following error: 

**0:00:13.292659217 32469       0xa79850 ERROR          interpipesink gstinterpipesink.c:803:gst_inter_pipe_sink_add_listener:<src_2> Can not add listener interpipesrc1 because our caps are not defined yet
0:00:13.292688259 32469       0xa79850 ERROR          interpipesink gstinterpipesink.c:803:gst_inter_pipe_sink_add_listener:<src_2> Can not add listener interpipesrc1 because our caps are not defined yet
0:00:13.292701290 32469       0xa79850 ERROR           interpipesrc gstinterpipesrc.c:380:gst_inter_pipe_src_start:<interpipesrc1> Could not listen to node src_2
0:00:13.292808152 32469       0xa79850 ERROR              gstdstate gstd_state.c:203:gstd_state_update:<GstdState@0x7f811408a820> Failed to change the state of the pipeline**

Please note, same caps are used in both the pipes and I have used the same videos(one copied with another name) in both the pipe so that caps will be same. First video will be played successfully, when switches to pipe2 source then it will give above error.

P.S. : Whats the point of stopping pipe3, If i am doing so it will close the sink and it will not be continuous playback.

**Comment by anilgond** [comment]
Hi @jcaballeros 

Its not working yet, I have tried above steps but I am getting following error:

**0:00:13.292659217 32469 0xa79850 ERROR interpipesink gstinterpipesink.c:803:gst_inter_pipe_sink_add_listener:<src_2> Can not add listener interpipesrc1 because our caps are not defined yet
0:00:13.292688259 32469 0xa79850 ERROR interpipesink gstinterpipesink.c:803:gst_inter_pipe_sink_add_listener:<src_2> Can not add listener interpipesrc1 because our caps are not defined yet
0:00:13.292701290 32469 0xa79850 ERROR interpipesrc gstinterpipesrc.c:380:gst_inter_pipe_src_start: Could not listen to node src_2
0:00:13.292808152 32469 0xa79850 ERROR gstdstate gstd_state.c:203:gstd_state_update:GstdState@0x7f811408a820 Failed to change the state of the pipeline**

Please note, same caps are used in both the pipes and I have used the same videos(one copied with another name) in both the pipe so that caps will be same. First video will be played successfully, when switches to pipe2 source then it will give above error.

P.S. : Whats the point of stopping pipe3, If i am doing so it will close the sink and it will not be continuous playback.

**Comment by jcaballeros** [comment]
Hi,

A fix for the error "Can not add listener interpipesrc1 because our caps are not defined yet" has been merged in release v1.1.3. Please update and test with the latest source code. Closing, please reopen if the issue persists.
